core/src/cluster_info.rs

```diff
@@ -614,16 +617,25 @@ impl ClusterInfo {
     ) {
         let now = timestamp();
         self.my_contact_info.write().unwrap().wallclock = now;
-        let entry =
-            CrdsValue::new_signed(CrdsData::ContactInfo(self.my_contact_info()), &self.keypair);
+        self.instance.write().unwrap().update_wallclock(now);
+        let entries: Vec<_> = vec![
+            CrdsData::ContactInfo(self.my_contact_info()),
+            CrdsData::NodeInstance(self.instance.read().unwrap().clone()),
```

Comment:
Do we need to send `NodeInstance` continually?   `CrdsData::Version` is just added once at startup

Reply:
That is a good question. Is there a risk that the value is not propagated if it is only pushed once?
Something like:
* The network is partitioned during the push process.
    * or, the value is [becomes outdated and not pushed any further](https://github.com/solana-labs/solana/blob/20c0e8793/core/src/crds_gossip_push.rs#L178-L188).
* and, the value expires and is purged from crds table, before it is returned in pull-requests.
    * or, similarly, it is [not processed in pull request responses because it is too old](https://github.com/solana-labs/solana/blob/f6a7732ff/core/src/crds_gossip_pull.rs#L339-L345).

There is an [`update_record_timestamp`](https://github.com/solana-labs/solana/blob/f6a7732ff/core/src/crds.rs#L240-L251) which extends `local-timestamp` of the associated values if we hear from a contact-info (on some but not all gossip paths), which may stop the node from getting purged from crds table as long as its contact info is propagated, but seems like if the `wallclock` is too old, it will not get propagated through gossip.

Reply:
> Is there a risk that the value is not propagated if it is only pushed once?

That sounds like it.  I guess version has this same problem then.  Maybe not a big deal here though, you're now more knowledgable in this area than I am so your call :)

Reply:
I am leaning towards sending these periodically, just in case they get dropped because of network partitions, message expiration, etc. It should not add too much overhead.

Reply:
sgtm